tensorflow_probability/examples/jupyter_notebooks/Probabilistic_Principal_Component_Analysis.ipynb

```diff
@@ -0,0 +1,444 @@
+    }
+   ],
+   "source": [
+    "plt.scatter(x_train[0, :], x_train[1, :], color='blue', alpha=0.1)\n",
+    "plt.axis('equal')\n",
+    "plt.axis([-10, 10, -10, 10])\n",
+    "plt.title(\"Simulated data set\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model\n",
+    "\n",
+    "Consider a data set $\\mathbf{X}=\\{\\mathbf{x}_{n}\\}$ of $N$ data points, where each data point is $D$-dimensional, $\\mathbf{x}_{n}\\in\\mathbb{R}^{D}$. We aim to represent each $\\mathbf{x}_{n}$ under a latent variable $\\mathbf{z}_{n}\\in\\mathbb{R}^{K}$ with lower dimension, $K<D$. The set of principal axes $\\mathbf{W}$ relates the latent variables to the data.\n",
+    "\n",
+    "Specifically, we assume that each latent variable is normally distributed,\n",
+    "$$\\mathbf{z}_{n}\\sim N(\\mathbf{0}, \\mathbf{I}).$$\n",
+    "\n",
+    "The corresponding data point is generated via a projection,\n",
+    "$$\\mathbf{x}_{n}\\mid\\mathbf{z}_{n}\\sim N(\\mathbf{W}\\mathbf{z}_{},\\sigma^{2}\\mathbf{I}),$$\n",
+    "\n",
+    "where the matrix $\\mathbf{W}\\in\\mathbb{R}^{D\\times K}$ are known as the principal axes. In probabilistic PCA, we are typically interested in estimating the principal axes $\\mathbf{W}$ and the noise term $\\sigma^{2}$.\n",
+    "\n",
+    "Probabilistic PCA generalizes classical PCA. Marginalizing out the latent variable, the distribution of each data point is\n",
+    "\n",
+    "$$\\mathbf{x}_{n}\\sim N(\\mathbf{0},\\mathbf{W}\\mathbf{W}^{T}+\\sigma^{2}\\mathbf{I}).$$\n",
+    "\n",
+    "Classical PCA is the specific case of probabilistic PCA when the covariance of the noise becomes infinitesimally small, $\\sigma^{2}\\to 0.$\n",
+    "We set up our model below. In our analysis, we fix $\\sigma=2.0$, and instead of point estimating $\\mathbf{W}$ as a model parameter, we place a prior over it in order to infer a distribution over principal axes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def ppca_model():\n",
+    "    \"\"\"Define Probabilistic PCA model.\"\"\"\n",
+    "    sigma = tf.constant(2.0, name='sigma')\n",
+    "    w = ed.Normal(loc=tf.zeros([D, K]),\n",
+    "                  scale=sigma*tf.ones([D, K]),\n",
+    "                  name='w')\n",
+    "    z = ed.Normal(loc=tf.zeros([N, K]),\n",
+    "                  scale=tf.ones([N, K]),\n",
+    "                  name='z')\n",
+    "    x = ed.Normal(loc=tf.matmul(w, z, transpose_b=True),\n",
+    "                  scale=tf.ones([D, N]),\n",
+    "                  name='x')\n",
+    "    return x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We visualize the model with TensorBoard."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def strip_consts(graph_def, max_const_size=32):\n",
```

Comment:
This is cool! Can you add a comment pointing to the original source of this code? (seems to be based on https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/deepdream/deepdream.ipynb ?)